services/src/main/java/org/keycloak/forms/login/freemarker/model/TotpBean.java

```diff
@@ -39,11 +44,17 @@
     private final String totpSecretQrCode;
     private final boolean enabled;
     private UriBuilder uriBuilder;
+    private final List<CredentialModel> otpCredentials;
 
     public TotpBean(KeycloakSession session, RealmModel realm, UserModel user, UriBuilder uriBuilder) {
         this.realm = realm;
         this.uriBuilder = uriBuilder;
-        this.enabled = session.userCredentialManager().isConfiguredFor(realm, user, OTPCredentialModel.TYPE);
+        this.enabled = ((OTPCredentialProvider)session.getProvider(CredentialProvider.class, "keycloak-otp")).isConfiguredFor(realm, user);
```

Comment:
Is it possible to keep the "this.enabled = session.userCredentialManager().isConfiguredFor(realm, user, OTPCredentialModel.TYPE);" ? Using of "userCredentialManager" instead of directly using OTPProvider is preferred way due the case when userStorage is used - some details in https://issues.redhat.com/browse/KEYCLOAK-12386

Reply:
Sure, thanks for catching this! Will change to the previous form (tested the result of both variants previously, and it seemed to be identical (maybe just in my particular testing case?), so changed it to the form it has had in the initial MFA change, KEYCLOAK-11745 one on some related place). But will change it to the original form too.

Reply:
@mposolda 
Fixed in the 2nd commit (intentionally kept it separate, so the changes from the original version are better visible). Will squash them into one, once the change is OK.